jupyter_server/services/sessions/sessionmanager.py

```diff
@@ -47,14 +46,12 @@ def _validate_database_filepath(self, proposal):
                 raise TraitError(
                     "`database_filepath` expected a file path, but the given path is a directory."
                 )
-            # If the file exists, but it's empty, its a valid entry.
-            if os.stat(path).st_size == 0:
-                return value
             # Verify that database path is an SQLite 3 Database by checking its header.
             with open(value, "rb") as f:
```

Comment:
I meant also for the contents validation piece, rather than inspecting the file

Reply:
I think `sqlite3` overwrites the file if it's not a valid database format, rather than raising an exception if the file has contents that are not sqlite data.

Reply:
Ah, yup, confirmed.